src/main/java/org/opensearch/knn/plugin/KNNEngineFactory.java

```diff
@@ -1,33 +0,0 @@
-/*
- * Copyright OpenSearch Contributors
- * SPDX-License-Identifier: Apache-2.0
- */
-
-package org.opensearch.knn.plugin;
-
-import org.opensearch.index.engine.Engine;
-import org.opensearch.index.engine.EngineConfig;
-import org.opensearch.index.engine.EngineFactory;
-import org.opensearch.index.engine.InternalEngine;
-
-/**
- * EngineFactory to inject the KNNCodecService to help segments write using the KNNCodec.
- */
-class KNNEngineFactory implements EngineFactory {
-
-    private static KNNCodecService codecService = new KNNCodecService();
-
-    @Override
-    public Engine newReadWriteEngine(EngineConfig config) {
-        codecService.setPostingsFormat(config.getCodec().postingsFormat());
```

Comment:
One call out: The only concern I have with this change is the removal of this line. I believe it will be irrelevant once #146 is fixed. @vamshin do you remember the reason for adding this here?

Reply:
Good catch. We would need this line otherwise it could be a breaking change. Reason for explicitly setting this is documented in PR https://github.com/opendistro-for-elasticsearch/k-NN/pull/236

Reply:
Moving PR into draft state for now. I think we can refactor this a little bit.